src/Sylius/Bundle/CoreBundle/EventListener/DefaultUsernameORMListener.php

```diff
@@ -35,13 +36,20 @@ public function onFlush(OnFlushEventArgs $onFlushEventArgs)
 
     private function processEntities(array $entities, EntityManagerInterface $entityManager, UnitOfWork $unitOfWork): void
     {
-        foreach ($entities as $customer) {
-            if (!$customer instanceof CustomerInterface) {
+        foreach ($entities as $entity) {
+            if (!$entity instanceof ShopUserInterface && !$entity instanceof CustomerInterface) {
```

Comment:
Can you maybe split this condition into two separate ones? The logic below could be extracted to private method with two arguments I suppose (`user` + `customer`), it should increase readability. Also, can you update specs?
```suggestion
            if ($entity instanceof ShopUserInterface) {
                $this->updateUserName($entity, $entity->getCustomer());
            }
            
            if ($entity instanceof CustomerInterface) {
                $this->updateUserName($entity->getUser(), $entity);
            }
```

Reply:
@lchrusciel 
so you think this is more readable ?
as i should also pass `$entityManager` and `$unitOfWork` to updateUserName
in addition to @var tag to bypass phpStan
```
private function processEntities(array $entities, EntityManagerInterface $entityManager, UnitOfWork $unitOfWork): void
    {
        foreach ($entities as $entity) {
            if ($entity instanceof ShopUserInterface) {
                /** @var CustomerInterface $customer */
                $customer = $entity->getCustomer();
                $this->updateUsername($entity, $customer, $entityManager, $unitOfWork);
            } elseif($entity instanceof CustomerInterface) {
                /** @var ShopUserInterface $user */
                $user = $entity->getUser();
                $this->updateUsername($user, $entity, $entityManager, $unitOfWork);
            }
        }
    }

private function updateUsername(?ShopUserInterface $user, ?CustomerInterface $customer, EntityManagerInterface $entityManager, UnitOfWork $unitOfWork): void
    {
        if (!$user || !$customer) {
            return;
        }

        if ($customer->getEmail() === $user->getUsername() && $customer->getEmailCanonical() === $user->getUsernameCanonical()) {
            return;
        }

        $user->setUsername($customer->getEmail());
        $user->setUsernameCanonical($customer->getEmailCanonical());

        /** @var ClassMetadata $userMetadata */
        $userMetadata = $entityManager->getClassMetadata(get_class($user));
        $unitOfWork->recomputeSingleEntityChangeSet($userMetadata, $user);
    }
```

Reply:
Yes, I would say, that it is easier to follow this logic because we have fewer conditions here. Don't you agree?